core/state_processor.go

```diff
@@ -494,6 +494,63 @@ func (p *StateProcessor) Process(block *types.WorkObject, batch ethdb.Batch) (ty
 							outputIndex++
 						}
 					}
+				} else if tx.To().IsInQuaiLedgerScope() { // Quai coinbase
+					internal, err := tx.To().InternalAndQuaiAddress()
+					if err != nil {
+						return nil, nil, nil, nil, 0, 0, 0, nil, nil, fmt.Errorf("coinbase tx %x has invalid recipient: %w", tx.Hash(), err)
+					}
+					code := statedb.GetCode(internal)
+					if len(code) == 0 && len(tx.Data()) == 1 {
+						// No code and coinbase has no extra data
+						// Coinbase is valid, no gas used
+						receipt = &types.Receipt{Type: tx.Type(), Status: types.ReceiptStatusSuccessful, GasUsed: 0, TxHash: tx.Hash()}
```

Comment:
coinbase uses gas right, its the 21000 default

Reply:
I'm pretty sure this is fixed, please check

Reply:
yeah its an old commit